src/icu.rs

```diff
@@ -54,11 +54,15 @@ pub fn set_common_data_71(data: &'static [u8]) -> Result<(), i32> {
 
 /// Returns BCP47 language tag.
 pub fn get_language_tag() -> String {
-  let mut output = [0u8; 1024];
-  let len = unsafe {
-    icu_get_default_locale(output.as_mut_ptr() as *mut c_char, output.len())
-  };
-  std::str::from_utf8(&output[..len]).unwrap().to_owned()
+  let mut output = Vec::<u8>::with_capacity(1024);
```

Comment:
You're not removing an allocation here, you're adding one.

Reply:
Can you elaborate?

Reply:
You're allocating a Vec.
Previously a the `[u8; 1024]` was living on the stack.

Reply:
"Allocating" on the stack is free. The only overhead here is filling the slide with zeroes (we could work around that by using MaybeUninit).

Reply:
> "Allocating" on the stack is free. The only overhead here is filling the slide with zeroes (we could work around that by using MaybeUninit).

Well, now the copy to the heap allocation is also eliminated.

Although, maybe we should measure the difference? (I'll comment on the Discord)

Reply:
> Well, now the copy to the heap allocation is also eliminated.

True, but in exchange the language tag now uses 1024 bytes on the heap instead of 10.